src/Groups/types.jl

```diff
@@ -256,6 +256,15 @@ Group of automorphisms over a group of type `T`. It can be defined via the funct
   end
 end
 
+(aut::AutomorphismGroup{T} where T)(x::GapObj) = group_element(aut,x)
+
+const AutomorphismGroupElem{GrpAbFinGen} = BasicGAPGroupElem{AutomorphismGroup{GrpAbFinGen}}
```

Comment:
This yields:
```julia
julia> AutomorphismGroupElem
Oscar.BasicGAPGroupElem{AutomorphismGroup{GrpAbFinGen}} where GrpAbFinGen
```
The `GrpAbFinGen` in your line is a free variable and does *not* refer to the type `GrpAbFinGen`. What you wrote is the same as 
```
julia> const F{T} = Oscar.BasicGAPGroupElem{AutomorphismGroup{T}}
Oscar.BasicGAPGroupElem{AutomorphismGroup{T}} where T
```
In other words, you defined `AutomorphismGroupElem{T}` to be an alias for `BasicGAPGroupElem{AutomorphismGroup{T}}` for all `T`. I am not sure this was the intention.

Reply:
Oh ok I see, so this should be changed as `const AutomorphismGroupElem{T} = BasicGAPGroupElem{AutomorphismGroup{T}} where T <: GrpAbFinGen` for instance ?

Reply:
This makes the name `AutomorphismGroupElem` unavailable for anything else, but we can do that for now.

Reply:
It also makes clear what the intention is.

Reply:
No but regarding how the structure `AutomorphismGroup` was defined, I guess we should allow `T` to be "whatever we want", just we will make clear, for now, that the methods are only meant for `T <: GrpAbFinGen`, as it has been implemented. Then I will just change the `GrpAbFinGen`'s by `T`'s.